update several sql statement docs (#2836)
pingcap/docs: sql-statements/sql-statement-show-character-set.md

## Synopsis

**ShowStmt:**
**ShowCharsetStmt:**

![ShowStmt](/media/sqlgram/ShowStmt.png)

**ShowTargetFilterable:**

![ShowTargetFilterable](/media/sqlgram/ShowTargetFilterable.png)

**CharsetKw:**

> ![CharsetKw](/media/sqlgram/CharsetKw.png)

Reviewer: These three lines should not be deleted according to https://github.com/pingcap/docs-cn/pull/3294/files.

## MySQL compatibility

> The usage of this statement is understood to be fully compatible with MySQL. However, charsets in TiDB may have different default collations compared with MySQL, the details can be found in [Compatibility with MySQL](/mysql-compatibility.md). Any other compatibility differences should be [reported via an issue](/report-issue.md) on GitHub.

Reviewer: ```suggestion
The usage of this statement is understood to be fully compatible with MySQL. However, charsets in TiDB may have different default collations compared with MySQL. For details, refer to [Compatibility with MySQL](/mysql-compatibility.md). Any other compatibility differences should be [reported via an issue](/report-issue.md) on GitHub.
```
